storage/storage.rs

```diff
@@ -159,42 +162,51 @@ impl<Durability> MVCCStorage<Durability> {
     }
 
     pub fn open_snapshot_write(self: Arc<Self>) -> WriteSnapshot<Durability> {
-        /*
-        How to pick a sequence number:
-
-        TXN 1 - open(0) ---> durably write = 10 PENDING ---> validate ---> write ---> committed. RETURN
-
-        Question: does external consistency make sense for single-node machines?
-        If user opens TXN 10 in thread 1, thread 2... work. User expects if open TXN after TXN 10 returns, we will see it. Otherwise, no expectations.
-        Therefore - we can always use the last committed state safely for happens-before relations.
-
-        For external consistency, we should use the the currently last pending sequence number and wait for it to finish.
-         */
-
-        let open_sequence_number = self.isolation_manager.watermark();
+        // guarantee external consistency: we always await the latest snapshots to finish
+        let possible_sequence_number = self.isolation_manager.highest_possible_watermark();
+        let mut open_sequence_number = self.snapshot_watermark();
+        while open_sequence_number < possible_sequence_number {
+            sleep(Duration::from_micros(WATERMARK_WAIT_INTERVAL_MICROSECONDS));
+            open_sequence_number = self.isolation_manager.watermark();
+        }
```

Comment:
This should be a function, e.g. `wait_for_watermark_to_reach(SequenceNumber) -> SequenceNumber`